docs/how-to/hip_porting_driver_api.rst

```diff
@@ -0,0 +1,480 @@
+.. meta::
+  :description: This chapter presents how to port the CUDA driver API and showcases equivalent operations in HIP.
+  :keywords: AMD, ROCm, HIP, CUDA, driver API
+
+.. _porting_driver_api:
+
+*******************************************************************************
+Porting CUDA driver API 
+*******************************************************************************
+
+NVIDIA provides separate CUDA Driver and Runtime APIs. The two APIs have significant overlap in functionality:
+
+* Both APIs support events, streams, memory management, memory copy, and error handling.
+* Both APIs deliver similar performance.
+* Driver API calls begin with the prefix ``cu``, while Runtime API calls begin with the prefix ``cuda``. For example, the Driver API contains ``cuEventCreate``, while the Runtime API contains ``cudaEventCreate``, which has similar functionality.
+* The Driver API defines a different, but largely overlapping, error code space than the Runtime API and uses a different coding convention. For example, the Driver API defines ``CUDA_ERROR_INVALID_VALUE``, while the Runtime API defines ``cudaErrorInvalidValue``.
+
+The Driver API offers two additional functionalities not provided by the Runtime API: ``cuModule`` and ``cuCtx`` APIs.
+
+cuModule API
+------------
+
+The Module section of the Driver API provides additional control over how and when accelerator code objects are loaded. For example, the driver API enables code objects to load from files or memory pointers. Symbols for kernels or global data are extracted from the loaded code objects. In contrast, the Runtime API loads automatically and, if necessary, compiles all the kernels from an executable binary when it runs. In this mode, kernel code must be compiled using NVCC so that automatic loading can function correctly.
```

Comment:
```suggestion
The Module section of the Driver API provides additional control over how and when accelerator code objects are loaded. For example, the Driver API enables code objects to load from files or memory pointers. Symbols for kernels or global data are extracted from the loaded code objects. In contrast, the Runtime API loads automatically and, if necessary, compiles all the kernels from an executable binary when it runs. In this mode, kernel code must be compiled using NVCC so that automatic loading can function correctly.
```

Reply:
Fixed.